web/src/main/java/se/inera/intyg/webcert/web/csintegration/certificate/CreateDraftCertificateFromCS.java

```diff
@@ -0,0 +1,115 @@
+import se.riv.clinicalprocess.healthcond.certificate.createdraftcertificateresponder.v3.Intyg;
+import se.riv.clinicalprocess.healthcond.certificate.v3.ErrorIdType;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class CreateDraftCertificateFromCS implements CreateDraftCertificate {
+
+    private final PatientDetailsResolver patientDetailsResolver;
+    private final CSIntegrationService csIntegrationService;
+    private final CSIntegrationRequestFactory csIntegrationRequestFactory;
+    private final IntegreradeEnheterRegistry integreradeEnheterRegistry;
+    private final PDLLogService pdlLogService;
+    private final MonitoringLogService monitoringLogService;
+
+    @Override
+    public CreateDraftCertificateResponseType create(Intyg certificate, IntygUser user) {
+        final var applicationError = validatePUIntegration(certificate);
+        if (applicationError.isPresent()) {
+            return applicationError.get();
+        }
+
+        final var modelId = csIntegrationService.certificateTypeExists(certificate.getTypAvIntyg().getCode());
+        if (modelId.isEmpty()) {
+            log.debug("Certificate type '{}' does not exist in certificate service", certificate.getTypAvIntyg().getCode());
+            return null;
+        }
+
+        integreradeEnheterRegistry.putIntegreradEnhet(getIntegreradEnhetEntry(user), false, true);
+
+        try {
+            final var createdCertificate = csIntegrationService.createCertificate(
+                csIntegrationRequestFactory.createDraftCertificateRequest(modelId.get(), certificate, user)
+            );
+
+            pdlLogService.logCreatedWithIntygUser(createdCertificate, user);
+            monitoringLogService.logUtkastCreated(
+                createdCertificate.getMetadata().getId(),
+                createdCertificate.getMetadata().getType(),
+                createdCertificate.getMetadata().getUnit().getUnitId(),
+                createdCertificate.getMetadata().getIssuedBy().getPersonId(),
+                0
+            );
+
+            return createSuccessResponse(createdCertificate.getMetadata().getId(), createdCertificate.getMetadata().getUnit().getUnitId());
+        } catch (Exception exception) {
+            return createErrorResponse(exception.getMessage(), ErrorIdType.VALIDATION_ERROR);
+        }
+    }
+
+    private static IntegreradEnhetEntry getIntegreradEnhetEntry(IntygUser user) {
+        return new IntegreradEnhetEntry(user.getValdVardenhet().getId(),
+            user.getValdVardenhet().getNamn(), user.getValdVardgivare().getId(), user.getValdVardgivare().getNamn());
+    }
+
+    private Optional<CreateDraftCertificateResponseType> validatePUIntegration(Intyg certificate) {
+        final var personIdExtension = certificate.getPatient().getPersonId().getExtension();
+        final var personId = Personnummer.createPersonnummer(personIdExtension)
+            .orElseThrow(() -> new IllegalArgumentException(
```

Comment:
Webcert throws WebCertServiceException if it is not possible. Use the same type of exception and type.

`        final var personnummer = Personnummer.createPersonnummer(
                certificate.getPatient().getPersonId().getExtension())
            .orElseThrow(() -> new WebCertServiceException(WebCertServiceErrorCodeEnum.PU_PROBLEM,
                "Failed to create valid personnummer for createDraft request"));`